brave/brave-core
# Migrate brave-core changes from Android Ads

Fixes https://github.com/brave/brave-browser/issues/5160

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

- Confirm ad notifications are shown
- Switch ads off and then back on and confirm ad notifications are still shown (when turning off ads, state is removed so you will need to visit a website for page classification)
- Confirm ads are cashed-in
- Confirm clicking an ad opens in a new tab (or same tab for duplicate ad)
- Confirm `default/ads_service` state is removed when ads are switched off
- Confirm `default/ads_service` state is created when ads are switched on

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
